pkg/imgpkg/signature/fetch_signatures_test.go

```diff
@@ -44,6 +45,34 @@ func TestSignatureRetriever_Signatures(t *testing.T) {
 		assert.Equal(t, imageset.UnprocessedImageRef{DigestRef: "registry.io/img@sha256:cf31af331f38d1d7158470e095b132acd126a7180a54f263d386da88eb681d93", Tag: "some-tag"}, sign2)
 	})
 
+	t.Run("denied errors are provided as part of the error", func(t *testing.T) {
+		fakeSignatureFinder := &signaturefakes.FakeFinder{}
+		subject := signature.NewSignatures(fakeSignatureFinder, 2)
+		fakeSignatureFinder.SignatureCalls(func(digest regname.Digest) (imageset.UnprocessedImageRef, error) {
+			availableResults := map[string]imageset.UnprocessedImageRef{
+				"sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0": {DigestRef: "registry.io/img@sha256:cf31af331f38d1d7158470e095b132acd126a7180a54f263d386da88eb681d93", Tag: "some-tag"},
+				"sha256:56cb33b3b4bc45509c5ff7513ddc6ed78764f9ad5165cc32826e04da49d5462b": {DigestRef: "registry.io/img2@sha256:be154cc2b1211a9f98f4d708f4266650c9129784d0485d4507d9b0fa05d928b6", Tag: "some-other-tag"},
+			}
+			if res, ok := availableResults[digest.DigestStr()]; ok {
+				return res, nil
+			}
+			return imageset.UnprocessedImageRef{}, signature.AccessDeniedErr{}
+		})
+
+		var args []lockconfig.ImageRef
+		args = append(args, lockconfig.ImageRef{Image: "registry.io/img@sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0"})
+		args = append(args, lockconfig.ImageRef{Image: "registry.io/img1@sha256:6716afd7a68262a37d3f67681ed9dedf3b882938ad777f268f44d68894531f7a"})
+		args = append(args, lockconfig.ImageRef{Image: "registry.io/img2@sha256:56cb33b3b4bc45509c5ff7513ddc6ed78764f9ad5165cc32826e04da49d5462b"})
+		signatures, err := subject.FetchForImageRefs(args)
+		require.Error(t, err)
+		errs, ok := err.(*signature.FetchError)
+		require.True(t, ok, "Unexpected error found %+v, while expecting a FetchError", err)
+		require.Len(t, errs.AllErrors, 1)
```

Comment:
is it worth this have >1 errors in it? Just to validate that when more than one access denied error occurs, that they are all shown/returned?

Reply:
The main reason for this check is the one after where we do
`require.EqualError(t, errs.AllErrors[0], "access denied")`
to make sure we do not panic or something.

Are you asking if we should have a test that contains more than 1 error?

Reply:
> Are you asking if we should have a test that contains more than 1 error?

This